### PR TITLE
Addresses #247 - Plaintext IMAP Connection Example

### PR DESCRIPTION
**Pull request by 641i130**
I found the documentation lacking in a simple plain text connection. It took me a bit to figure it out but I did (I'm a Rust beginner after all). I'm hoping others will find this useful.
(Addressing #247)

- - -
This change is [Reviewable](https://reviewable.io/reviews/jonhoo/rust-imap/248)

### CODE REVIEW

File: examples/README.md [hunk 1]
```diff
@@ -1,12 +1,13 @@
...
-  * starttls - This is an example showing how to use STARTTLS after connecting over plaintext.
-  * timeout - This demonstrates how to use timeouts while connecting to an IMAP server by using a custom TCP/TLS stream initialization and creating a `Client` directly instead of using the `ClientBuilder`.
+
+- basic - This is a very basic example of using the client.
+- gmail_oauth2 - This is an example using oauth2 for logging into gmail as a secure appplication.
+- idle - This is an example showing how to use IDLE to monitor a mailbox.
+- rustls - This demonstrates how to use Rustls instead of Openssl for secure connections (helpful for cross compilation).
+- starttls - This is an example showing how to use STARTTLS after connecting over plaintext.
+- timeout - This demonstrates how to use timeouts while connecting to an IMAP server by using a custom TCP/TLS stream initialization and creating a `Client` directly instead of using the `ClientBuilder`.
+- plaintext - This demonstrates how to make an unencrypted IMAP connection (usually over 143) with a `Client` using a naked TCP connection.
```
Comment (jonhoo) [inline examples/README.md:13]: How come you changed the formatting of this list?

File: examples/plaintext.rs [hunk 2]
```diff
@@ -0,0 +1,54 @@
+use std::net::TcpStream;
+
+fn main() {
+    // REMINDER this is unsafe, the credentials are sent over the connection in CLEARTEXT
```
Comment (jonhoo) [inline examples/plaintext.rs:4]: The word "unsafe" has a very particular meaning in the Rust ecosystem. I suggest we go with "dangerous" instead.

File: examples/plaintext.rs [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        Err(e) => {
+            eprintln!("Connection error!");
+            eprintln!("{:?}",e);
+        }
+    }
+}
+
+fn plaintext() -> imap::error::Result<Option<String>> {
+    // Make a raw TCP connection to an UNSAFE IMAP server
+    let stream = TcpStream::connect("imap.example.com:143").unwrap(); // This is unsafe.
```
Comment (jonhoo) [inline examples/plaintext.rs:21]: Same here: s/unsafe/dangerous/ or "risky" maybe?

Comment (jonhoo) [reply examples/plaintext.rs:21]: Also, it's reallt the `.login` line that's risky, not this one.

File: examples/plaintext.rs [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    // extract the message's body
+    let mut body;
+    match message.body() {
+        Some(msg) => {
+            body = std::str::from_utf8(msg)
+                .expect("message was not valid utf-8")
+                .to_string();
+        }
+        None => body = "".to_string(),
+    }
```
Comment (jonhoo) [inline examples/plaintext.rs:50]:
```suggestion
    let body = message.body().map(|body| {
        String::from_utf8(body).expect("message was not valid utf-8")
    }).unwrap_or_else(String::new);
```

**Review by jonhoo — CHANGES_REQUESTED** [review]
Adding an example for this seems pretty reasonable! I left a few notes inline :)

File: examples/README.md (hunk 1 above)
Comment (641i130) [reply examples/README.md:13]: Ah it seems my text editor updated it oddly. Sorry about that, I'll revert it back to how it was!

**Comment by 641i130** [comment]
I updated all the changes requested. Let me know if I did that correctly. I've never done this through GitHub before!

File: examples/plaintext.rs [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+        }
+        Err(e) => {
+            eprintln!("Connection error!");
+            eprintln!("{:?}", e);
+        }
+    }
+}
+
+fn plaintext() -> imap::error::Result<Option<String>> {
+    let stream = TcpStream::connect("imap.example.com:143").unwrap(); // This is unsafe.
```
Comment (jonhoo) [inline examples/plaintext.rs:20]: This still says "This is unsafe" at the end, which I think we can now remove.

File: examples/plaintext.rs [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+    let mut client = imap::Client::new(stream);
+    client.read_greeting()?;
+    eprintln!("\nUNENCRYPTED connection made!!!!\n");
+    eprintln!("This is highly not recommended.\n");
+    // to do anything useful with the e-mails, we need to log in
+
+    // Makes an unencrypted login message to the IMAP server. This is risky business!
```
Comment (jonhoo) [inline examples/plaintext.rs:27]: The formatting here is a little odd. Let's be consistent about upper-casing in the comments (all lower-case is fine), and maybe remove the empty line?

I'd also move the risky business comment to its own line. Like so:
```suggestion
    // to do anything useful with the e-mails, we need to log in
    // keep in mind that this is over plain TCP, so may leak all your secrets!
```

File: examples/plaintext.rs [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+        }
+        Err(e) => {
+            eprintln!("Connection error!");
+            eprintln!("{:?}", e);
+        }
+    }
+}
+
+fn plaintext() -> imap::error::Result<Option<String>> {
+    let stream = TcpStream::connect("imap.example.com:143").unwrap(); // This is an unencrypted connection.
```
Comment (jonhoo) [inline examples/plaintext.rs:20]:
```suggestion
    let stream = TcpStream::connect("imap.example.com:143").unwrap();
```

**Review by jonhoo — APPROVED** [review]

**Comment by jonhoo** [comment]
Looks good! Now we just need to make sure it compiles :sweat_smile:

**Comment by urkle** [comment]
@jonhoo note this may not be needed once (or at least significantly simpler) we land #245 as making a plain text connection becomes simpler with the new ClientBuilder

**Review by jonhoo — APPROVED** [review]
Excellent, thank you :tada:

**Pull request merged**